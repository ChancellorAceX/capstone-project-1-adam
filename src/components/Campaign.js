import React from 'react';
import EncounterListItem from './EncounterListItem';
import DeleteButton from './DeleteButton';

export default function Campaign(props) {
  return (
    <section className='campaignListing'>
      <h4 className='campaignTitle'>{props.campaign[1]}
        <DeleteButton
          target='campaign'
          id={+props.campaign[0]}
          campaigns={props.campaigns}
          setTargetState={props.setTargetState}
        />
      </h4>
      <ul className='campaignListing'>
        {props.encounters.map(encounter =>
          <EncounterListItem
            key={encounter.eid}
            encounter={encounter}
            encounters={props.encountersSource}
            setTargetState={props.setTargetState}
          />)}
      </ul>
    </section>
  );
}