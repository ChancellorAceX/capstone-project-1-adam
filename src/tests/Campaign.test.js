import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Campaign from '../components/Campaign';

describe('Campaign component', () => {
  const props = {
    "campaigns": [
      "[1, \"Azeroth\"]",
      "[6, \"Fillory\"]",
      "[7, \"Grishaverse\"]"
    ],
    "campaign": [
      1,
      "Azeroth"
    ],
    "encountersSource": [
      "{campaignuserid: 1, cid: 1, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 6, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 7, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 1, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 1, created: \"2020-07-03T00…}"
    ],
    "encounters": [
      "{campaignuserid: 1, cid: 1, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 1, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 1, created: \"2020-07-03T00…}"
    ],
    "setTargetState": "ƒ () {}"
  };

  it('renders the full campaign display unit', () => {
    const wrapper = shallow(<Campaign {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});