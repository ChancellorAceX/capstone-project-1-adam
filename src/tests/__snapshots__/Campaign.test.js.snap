// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campaign component renders the full campaign display unit 1`] = `
<section
  className="campaignListing"
>
  <h4
    className="campaignTitle"
  >
    Azeroth
    <DeleteButton
      campaigns={
        Array [
          "[1, \\"Azeroth\\"]",
          "[6, \\"Fillory\\"]",
          "[7, \\"Grishaverse\\"]",
        ]
      }
      id={1}
      setTargetState="ƒ () {}"
      target="campaign"
    />
  </h4>
  <ul
    className="campaignListing"
  >
    <EncounterListItem
      encounter="{campaignuserid: 1, cid: 1, created: \\"2020-07-03T00…}"
      encounters={
        Array [
          "{campaignuserid: 1, cid: 1, created: \\"2020-07-03T00…}",
          "{campaignuserid: 1, cid: 6, created: \\"2020-07-03T00…}",
          "{campaignuserid: 1, cid: 7, created: \\"2020-07-03T00…}",
          "{campaignuserid: 1, cid: 1, created: \\"2020-07-03T00…}",
          "{campaignuserid: 1, cid: 1, created: \\"2020-07-03T00…}",
        ]
      }
      setTargetState="ƒ () {}"
    />
    <EncounterListItem
      encounter="{campaignuserid: 1, cid: 1, created: \\"2020-07-03T00…}"
      encounters={
        Array [
          "{campaignuserid: 1, cid: 1, created: \\"2020-07-03T00…}",
          "{campaignuserid: 1, cid: 6, created: \\"2020-07-03T00…}",
          "{campaignuserid: 1, cid: 7, created: \\"2020-07-03T00…}",
          "{campaignuserid: 1, cid: 1, created: \\"2020-07-03T00…}",
          "{campaignuserid: 1, cid: 1, created: \\"2020-07-03T00…}",
        ]
      }
      setTargetState="ƒ () {}"
    />
    <EncounterListItem
      encounter="{campaignuserid: 1, cid: 1, created: \\"2020-07-03T00…}"
      encounters={
        Array [
          "{campaignuserid: 1, cid: 1, created: \\"2020-07-03T00…}",
          "{campaignuserid: 1, cid: 6, created: \\"2020-07-03T00…}",
          "{campaignuserid: 1, cid: 7, created: \\"2020-07-03T00…}",
          "{campaignuserid: 1, cid: 1, created: \\"2020-07-03T00…}",
          "{campaignuserid: 1, cid: 1, created: \\"2020-07-03T00…}",
        ]
      }
      setTargetState="ƒ () {}"
    />
  </ul>
</section>
`;
